src/bin/cargo/commands/owner.rs

```diff
@@ -6,25 +6,68 @@ use cargo::util::auth::Secret;
 pub fn cli() -> Command {
     subcommand("owner")
         .about("Manage the owners of a crate on the registry")
-        .arg_quiet()
         .arg(Arg::new("crate").action(ArgAction::Set))
-        .arg(
-            multi_opt(
-                "add",
-                "LOGIN",
-                "Name of a user or team to invite as an owner",
-            )
-            .short('a'),
-        )
-        .arg(
-            multi_opt(
-                "remove",
-                "LOGIN",
-                "Name of a user or team to remove as an owner",
-            )
-            .short('r'),
+        .arg_required_else_help(true)
+        .override_usage(
+            "\
+       cargo owner [OPTIONS] add    OWNER_NAME <CRATE_NAME>
+       cargo owner [OPTIONS] remove OWNER_NAME <CRATE_NAME>
+       cargo owner [OPTIONS] list   <CRATE_NAME>",
         )
-        .arg(flag("list", "List owners of a crate").short('l'))
+        .arg_quiet()
+        .subcommands([
+            Command::new("add")
+                .about("Name of a user or team to invite as an owner")
+                .override_usage(
+                    "\
+                   cargo owner [OPTIONS] add [OWNER_NAME] <CRATE_NAME>",
+                )
+                .arg_quiet()
+                .args([
+                    Arg::new("ownername")
+                        .action(ArgAction::Set)
+                        .required(true)
+                        .num_args(1)
+                        .value_delimiter(',')
+                        .value_name("OWNER_NAME"),
+                    Arg::new("cratename")
```

Comment:
I know ehuss mentioned making this a positional but I wonder if it should be an option instead, especially now that its optional.  I see positionals like function parameters (in languages with named parameters): if the meaning isn't obvious by the position then it should be a named parameter.

Benefits
- Clearer meaning
- `ownername` can use `num_args(1..)`
- Might be less confusing than a straight parameter order swap

Reply:
Do you mean to change ownername and crate name to options? I want to discuss：

1. Currently, it is supported to use ```.value_delimiter(',')``` to add or delete multiple owners at one time;
2. If ```ownername``` and ```cratename``` are optional, ```cargo owner add/remove``` will have no meaning;

Reply:
I was thinking owners be a required positional and package name be an option that gets defaulted to the one for the parent directory.

Reply:
Now, crate name is an option.